### DISCUSSION TITLE
Dynamic answer options in drop down of questionnaire

### DISCUSSION THREAD

**Discussion by mansikalraa** [google/android-fhir#2227] [Q&A] [answered]
Hi, Can anyone help me with populating dropdown answer options in questionnaire with data retrieved from some other source, i.e., the answer options are dynamic and not known before.

@jingtang10 @fredhersch

**Comment by MJ1998** [comment, marked as answer]
Hi @mansikalraa 

In short, you can use one of the configuration option in `DataCaptureConfig`: `valueSetResolverExternal`. So you would need to implement `resolve()` which takes in the custom value set uri and gives back a List\<Coding\> which is populated as answer options. Check [kdoc](https://github.com/google/android-fhir/blob/c26f6545c969c6f4946db555deecca4412bdc24a/datacapture/src/main/java/com/google/android/fhir/datacapture/DataCaptureConfig.kt#L40) for more information.

Let us know if you still need help with this.

**Reply by mansikalraa** [reply]
Thank you.
